jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -385,6 +419,205 @@ public Integer getMaxQueueSize() {
+     * Creates a new configuration based on the given sender.
+     * @param sender the sender to use
+     */
+    public SenderConfiguration(Sender sender) {
+      this.sender = sender;
+    }
+
+    /**
+     * Creates a new Sender configuration sending data to an agent via UDP
+     * @param agentHost the agent's hostname
+     * @param agentPort the agent's port
+     */
+    public SenderConfiguration(String agentHost, Integer agentPort) {
+      this(agentHost, agentPort, null, null, null, null);
+    }
+
+    /**
+     * Creates a new Sender configuration sending data to an endpoint, passing an authentication bearer token
+     * @param endpoint  the endpoint
+     * @param authToken the authentication token to be added as bearer token
+     */
+    public SenderConfiguration(String endpoint, String authToken) {
+      this(null, null, endpoint, authToken, null, null);
+    }
+
+    /**
+     * Creates a new Sender configuration sending data to an endpoint, passing the username and password as
+     * HTTP Basic Authentication
+     * @param endpoint     the endpoint
+     * @param authUsername the HTTP Basic Auth username
+     * @param authPassword the HTTP Basic Auth password
+     */
+    public SenderConfiguration(String endpoint, String authUsername, String authPassword) {
+      this(null, null, endpoint, null, authUsername, authPassword);
+    }
+
+    private SenderConfiguration(String agentHost, Integer agentPort, // agent-related properties
+                                String endpoint, // only required property when using the collector
+                                String authToken, // one valid auth combination
+                                String authUsername, String authPassword) { // the other valid auth
+      this.agentHost = agentHost;
+      this.agentPort = agentPort;
+      this.endpoint = endpoint;
+      this.authToken = authToken;
+      this.authUsername = authUsername;
+      this.authPassword = authPassword;
+    }
+
+    /**
+     * Returns a sender if one was given when creating the configuration, or attempts to create a sender based on the
+     * configuration's state.
+     * @return the sender passed via the constructor or a properly configured sender
+     */
+    public Sender getSender() {
+      // if we have a sender, that's the one we return
+      if (null != this.sender) {
+        return this.sender;
+      }
+
+      if (null != endpoint && !endpoint.isEmpty()) {
+        log.debug("The endpoint is set. Attempting to configure HttpSender for it.");
+        Interceptor authInterceptor = null;
+
+        // first, try to use token, if the env var is specified
+        if (null != authToken && !authToken.isEmpty()) {
+          log.debug("Auth Token environment variable found.");
+          authInterceptor = getAddTokenInterceptor(authToken);
+        }
+
+        // this will override the token if both values are specified
+        if (null != authUsername && !authUsername.isEmpty()
```

Comment:
This should be moved above the authToken if statement and use if/else - rather than creating the authInterceptor for authToken and overriding it with basic auth version if both sets of properties specified.

Reply:
I changed it, but I think it's more readable the other way. Regarding performance, it shouldn't matter much, as this code is executed only once.